src/nimony/sem.nim

```diff
@@ -7121,6 +7121,55 @@ proc writeOutput(c: var SemContext; outfile: string) =
       toBuild: move c.toBuild,
       exportBuf: buildIndexExports(c))
 
+  # Update .2.deps.nif file that doesn't contain modules imported under `when false:`
+  # so that Hexer and following phases doesn't read such modules.
+  var deps = createTokenBuf(16)
+  deps.buildTree StmtsS, NoLineInfo:
+    if c.importedModules.len != 0:
+      let systemSymId = if {SkipSystem, IsSystem} * c.moduleFlags == {}:
+                          c.currentScope.tab[pool.strings.getOrIncl("system")][0].name
+                        else:
+                          SymId(0)
+      deps.buildTree ImportS, NoLineInfo:
+        for symId, i in c.importedModules:
+          if symId != systemSymId:
+            deps.addStrLit i.path.toAbsolutePath
+  let depsFile = changeFileExt(outfile, ".deps.nif")
+  writeFile depsFile, "(.nif24)\n" & toString(deps)
+
+proc quitWithError(c: var SemContext; errors: sink TokenBuf = createTokenBuf(16)) =
+  # See https://github.com/nim-lang/nimony/issues/985#issuecomment-2849271319
+  if {IsMain, IsSystem} * c.moduleFlags != {}:
+    quit 1
+  else:
+    writeFile c.outfile, "\n"   # nifreader fail to open if it is empty.
+    var errs = ensureMove errors
+    var n = beginRead c.dest
+    errs.buildTree ErrT, NoLineInfo:
+      var nested = 0
+      while true:
+        case n.kind
+        of ParLe:
+          if n.tagId == ErrT:
+            errs.takeTree n
+            if nested == 0:
+              break
+            continue
```

Comment:
1. I don't accept break+continue dances. Remove the continue and duplicate the `inc n` step instead where needed.
2. Breaking in the ParLe case when nested == 0 as opposed to the ParRi case is never done elsewhere in the entire codebase so I don't understand it. I also don't want to understand it, rewrite it so that it is obvious what it does.